packages/cardano-services/src/StakePool/TypeormStakePoolProvider/TypeormStakePoolProvider.ts

```diff
@@ -72,28 +132,62 @@ export class TypeormStakePoolProvider extends TypeormProvider implements StakePo
       );
     }
 
-    this.logger.debug('About to query projected stake pools');
-
-    const { clause, args } = getWhereClauseAndArgs(filters);
-    const { field, order } = getSortOptions(sort);
+    // eslint-disable-next-line complexity
+    return this.withDataSource(async (dataSource: DataSource) => {
+      const queryRunner = dataSource.createQueryRunner();
+
+      let rawResult: PoolModel[];
+      let sortByScore = false;
+      let textFilter = false;
+
+      try {
+        if (withTextFilter(filters)) {
+          sortByScore = sort === undefined;
+          textFilter = true;
+
+          const values = (await this.getFuse(dataSource, fuzzyOptions))
+            .search(filters.text)
+            .map(({ item: { pool_id }, score }) => `('${pool_id}',${score})`)
+            .join(',');
+
+          await queryRunner.query(
+            [
+              'DROP TABLE IF EXISTS tmp_fuzzy',
```

Comment:
Temporary tables are per session: many connections can create many temporary tables with the same name without interfering with each other

Reply:
Sorry, with session I meant a DB session, each new db connection (in **TypeORM**, a `queryRunner`) is a new session.

Each session

1. drops the table... just in case
2. creates the temporary table and populates it
3. uses it one shot
4. even if in case of errors, it tries to drop the table

Reply:
But it's probably using a connection pool, does it re-create the session when it picks up an existing connection from the pool? :thinking: